# openapi-java-client

Rat Genome Database REST API
- API version: 1.1
  - Build date: 2024-10-12T11:25:44.365518-04:00[America/New_York]
  - Generator version: 7.9.0

The RGD REST API provides programmatic access to information and annotation stored in the Rat Genome Database

  For more information, please visit [http://rgd.mcw.edu](http://rgd.mcw.edu)

*Automatically generated by the [OpenAPI Generator](https://openapi-generator.tech)*


## Requirements

Building the API client library requires:
1. Java 1.8+
2. Maven (3.8.3+)/Gradle (7.2+)

## Installation

To install the API client library to your local Maven repository, simply execute:

```shell
mvn clean install
```

To deploy it to a remote Maven repository instead, configure the settings of the repository and execute:

```shell
mvn clean deploy
```

Refer to the [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html) for more information.

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>org.openapitools</groupId>
  <artifactId>openapi-java-client</artifactId>
  <version>1.1</version>
  <scope>compile</scope>
</dependency>
```

### Gradle users

Add this dependency to your project's build file:

```groovy
  repositories {
    mavenCentral()     // Needed if the 'openapi-java-client' jar has been published to maven central.
    mavenLocal()       // Needed if the 'openapi-java-client' jar has been published to the local maven repo.
  }

  dependencies {
     implementation "org.openapitools:openapi-java-client:1.1"
  }
```

### Others

At first generate the JAR by executing:

```shell
mvn clean package
```

Then manually install the following JARs:

* `target/openapi-java-client-1.1.jar`
* `target/lib/*.jar`

## Getting Started

Please follow the [installation](#installation) instruction and execute the following Java code:

```java

// Import classes:
import org.openapitools.client.ApiClient;
import org.openapitools.client.ApiException;
import org.openapitools.client.Configuration;
import org.openapitools.client.model.*;
import org.openapitools.client.api.AgrApi;

public class Example {
  public static void main(String[] args) {
    ApiClient defaultClient = Configuration.getDefaultApiClient();
    defaultClient.setBasePath("http://rest.rgd.mcw.edu/rgdws");

    AgrApi apiInstance = new AgrApi(defaultClient);
    String taxonId = "taxonId_example"; // String | The taxon ID for species
    try {
      Map<String, Object> result = apiInstance.getAffectedGenomicModelsUsingGET(taxonId);
      System.out.println(result);
    } catch (ApiException e) {
      System.err.println("Exception when calling AgrApi#getAffectedGenomicModelsUsingGET");
      System.err.println("Status code: " + e.getCode());
      System.err.println("Reason: " + e.getResponseBody());
      System.err.println("Response headers: " + e.getResponseHeaders());
      e.printStackTrace();
    }
  }
}

```

## Documentation for API Endpoints

All URIs are relative to *http://rest.rgd.mcw.edu/rgdws*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AgrApi* | [**getAffectedGenomicModelsUsingGET**](docs/AgrApi.md#getAffectedGenomicModelsUsingGET) | **GET** /agr/affectedGenomicModels/{taxonId} | Get affected genomic models (rat strains with gene alleles) submitted by RGD to AGR by taxonId
*AgrApi* | [**getAllelesForTaxonUsingGET**](docs/AgrApi.md#getAllelesForTaxonUsingGET) | **GET** /agr/alleles/{taxonId} | Get gene allele records submitted by RGD to AGR by taxonId
*AgrApi* | [**getExpressionForTaxonUsingGET**](docs/AgrApi.md#getExpressionForTaxonUsingGET) | **GET** /agr/expression/{taxonId} | Get expression annotations submitted by RGD to AGR by taxonId
*AgrApi* | [**getGenesForLatestAssemblyUsingGET**](docs/AgrApi.md#getGenesForLatestAssemblyUsingGET) | **GET** /agr/{taxonId} | Get gene records submitted by RGD to AGR by taxonId
*AgrApi* | [**getPhenotypesForTaxonUsingGET**](docs/AgrApi.md#getPhenotypesForTaxonUsingGET) | **GET** /agr/phenotypes/{taxonId} | Get phenotype annotations submitted by RGD to AGR by taxonId
*AgrApi* | [**getVariantsForTaxonUsingGET**](docs/AgrApi.md#getVariantsForTaxonUsingGET) | **GET** /agr/variants/{taxonId} | Get basic variant records submitted by RGD to AGR by taxonId
*AnnotationApi* | [**getAnnotationCountByAccIdAndObjectTypeUsingGET**](docs/AnnotationApi.md#getAnnotationCountByAccIdAndObjectTypeUsingGET) | **GET** /annotations/count/{accId}/{speciesTypeKey}/{includeChildren}/{objectType} | Returns annotation count for ontology accession ID and object type
*AnnotationApi* | [**getAnnotationCountByAccIdAndSpeciesUsingGET**](docs/AnnotationApi.md#getAnnotationCountByAccIdAndSpeciesUsingGET) | **GET** /annotations/count/{accId}/{speciesTypeKey}/{includeChildren} | Returns annotation count for ontology accession ID and speicies
*AnnotationApi* | [**getAnnotationCountByAccIdUsingGET**](docs/AnnotationApi.md#getAnnotationCountByAccIdUsingGET) | **GET** /annotations/count/{accId}/{includeChildren} | Returns annotation count for ontology accession ID
*AnnotationApi* | [**getAnnotationsByAccIdAndRgdIdUsingGET**](docs/AnnotationApi.md#getAnnotationsByAccIdAndRgdIdUsingGET) | **GET** /annotations/{accId}/{rgdId} | Returns a list of annotations by RGD ID and ontology term accession ID
*AnnotationApi* | [**getAnnotationsByRgdIdAndOntologyUsingGET**](docs/AnnotationApi.md#getAnnotationsByRgdIdAndOntologyUsingGET) | **GET** /annotations/rgdId/{rgdId}/{ontologyPrefix} | Returns a list of annotations by RGD ID and ontology prefix
*AnnotationApi* | [**getAnnotationsByRgdIdUsingGET**](docs/AnnotationApi.md#getAnnotationsByRgdIdUsingGET) | **GET** /annotations/rgdId/{rgdId} | Returns a list of annotations by RGD ID
*AnnotationApi* | [**getAnnotationsUsingGET**](docs/AnnotationApi.md#getAnnotationsUsingGET) | **GET** /annotations/{accId}/{speciesTypeKey}/{includeChildren} | Returns a list annotations for an ontology term or a term and it&#39;s children
*AnnotationApi* | [**getAnnotationsUsingPOST**](docs/AnnotationApi.md#getAnnotationsUsingPOST) | **POST** /annotations/ | Return a list of genes annotated to an ontology term
*AnnotationApi* | [**getAnnotsByRefrerenceUsingGET**](docs/AnnotationApi.md#getAnnotsByRefrerenceUsingGET) | **GET** /annotations/reference/{refRgdId} | Returns a list of annotations for a reference
*AnnotationApi* | [**getTermAccIdsUsingGET**](docs/AnnotationApi.md#getTermAccIdsUsingGET) | **GET** /annotations/accId/{rgdId} | Returns a list ontology term accession IDs annotated to an rgd object
*ChromosomeApi* | [**getChromosomeByAssemblyUsingGET**](docs/ChromosomeApi.md#getChromosomeByAssemblyUsingGET) | **GET** /maps/chr/{chromosome}/{mapKey} | Return a list of chromosomes
*ChromosomeApi* | [**getChromosomesByAssemblyUsingGET**](docs/ChromosomeApi.md#getChromosomesByAssemblyUsingGET) | **GET** /maps/chr/{mapKey} | Return a list of chromosomes
*EnrichmentWebServiceApi* | [**getEnrichmentDataUsingPOST**](docs/EnrichmentWebServiceApi.md#getEnrichmentDataUsingPOST) | **POST** /enrichment/annotatedGenes | Return a list of genes annotated to the term.Genes are rgdids separated by comma.Species type is an integer value.term is the ontology
*EnrichmentWebServiceApi* | [**getEnrichmentDataUsingPOST1**](docs/EnrichmentWebServiceApi.md#getEnrichmentDataUsingPOST1) | **POST** /enrichment/data | Return a chart of ontology terms annotated to the genes.Genes are rgdids separated by comma.Species type is an integer value.Aspect is the Ontology group
*GeneApi* | [**getAllAnnotatedGenesUsingGET**](docs/GeneApi.md#getAllAnnotatedGenesUsingGET) | **GET** /genes/annotation/{accId} | Return a list of genes annotated to an ontology term
*GeneApi* | [**getAnnotatedGenesUsingPOST**](docs/GeneApi.md#getAnnotatedGenesUsingPOST) | **POST** /genes/annotation | Return a list of genes annotated to an ontology term
*GeneApi* | [**getGeneAllelesUsingGET**](docs/GeneApi.md#getGeneAllelesUsingGET) | **GET** /genes/allele/{rgdId} | Return a list of gene alleles
*GeneApi* | [**getGeneByMapKeyUsingGET**](docs/GeneApi.md#getGeneByMapKeyUsingGET) | **GET** /genes/map/{mapKey} | Return a list of all genes with position information for an assembly
*GeneApi* | [**getGeneByRgdIdUsingGET**](docs/GeneApi.md#getGeneByRgdIdUsingGET) | **GET** /genes/{rgdId} | Get a gene record by RGD ID
*GeneApi* | [**getGeneBySymbolUsingGET**](docs/GeneApi.md#getGeneBySymbolUsingGET) | **GET** /genes/{symbol}/{speciesTypeKey} | Get a gene record by symbol and species type key
*GeneApi* | [**getGeneOrthologsUsingGET**](docs/GeneApi.md#getGeneOrthologsUsingGET) | **GET** /genes/orthologs/{rgdId} | Return a list of gene orthologs
*GeneApi* | [**getGenesAnnotatedUsingGET**](docs/GeneApi.md#getGenesAnnotatedUsingGET) | **GET** /genes/annotation/{accId}/{speciesTypeKey} | Return a list of genes annotated to an ontology term
*GeneApi* | [**getGenesByAffyIdUsingGET**](docs/GeneApi.md#getGenesByAffyIdUsingGET) | **GET** /genes/affyId/{affyId}/{speciesTypeKey} | Return a list of genes for an affymetrix ID
*GeneApi* | [**getGenesByAliasSymbolUsingGET**](docs/GeneApi.md#getGenesByAliasSymbolUsingGET) | **GET** /genes/alias/{aliasSymbol}/{speciesTypeKey} | Return a list of genes for an alias and species
*GeneApi* | [**getGenesByKeywordUsingGET**](docs/GeneApi.md#getGenesByKeywordUsingGET) | **GET** /genes/keyword/{keyword}/{speciesTypeKey} | Return a list of genes by keyword and species type key
*GeneApi* | [**getGenesByPositionUsingGET**](docs/GeneApi.md#getGenesByPositionUsingGET) | **GET** /genes/{chr}/{start}/{stop}/{mapKey} | Return a list of genes position and map key
*GeneApi* | [**getGenesBySpeciesUsingGET**](docs/GeneApi.md#getGenesBySpeciesUsingGET) | **GET** /genes/species/{speciesTypeKey} | Return a list of all genes for a species in RGD
*GeneApi* | [**getGenesInRegionUsingGET**](docs/GeneApi.md#getGenesInRegionUsingGET) | **GET** /genes/region/{chr}/{start}/{stop}/{mapKey} | Return a list of genes in region
*GeneApi* | [**getMappedGenesByPositionUsingGET**](docs/GeneApi.md#getMappedGenesByPositionUsingGET) | **GET** /genes/mapped/{chr}/{start}/{stop}/{mapKey} | Return a list of genes position and map key
*GeneApi* | [**getOrthologsByListUsingPOST**](docs/GeneApi.md#getOrthologsByListUsingPOST) | **POST** /genes/orthologs | Return a list of gene orthologs
*LookupApi* | [**getEnsemblGeneMappingUsingGET**](docs/LookupApi.md#getEnsemblGeneMappingUsingGET) | **GET** /lookup/id/map/EnsemblGene/{rgdId} | Translate an RGD ID to an Ensembl Gene  ID
*LookupApi* | [**getEnsemblGeneMappingUsingPOST**](docs/LookupApi.md#getEnsemblGeneMappingUsingPOST) | **POST** /lookup/id/map/EnsemblGene | Translate RGD IDs to Ensembl Gene IDs
*LookupApi* | [**getEnsemblProteinMappingUsingGET**](docs/LookupApi.md#getEnsemblProteinMappingUsingGET) | **GET** /lookup/id/map/EnsemblProtein/{rgdId} | Translate an RGD ID to an Ensembl Protein ID
*LookupApi* | [**getEnsemblProteinMappingUsingPOST**](docs/LookupApi.md#getEnsemblProteinMappingUsingPOST) | **POST** /lookup/id/map/EnsemblProtein | Translate RGD IDs to Ensembl Protein IDs
*LookupApi* | [**getEnsemblTranscriptMappingUsingGET**](docs/LookupApi.md#getEnsemblTranscriptMappingUsingGET) | **GET** /lookup/id/map/EnsemblTranscript/{rgdId} | Translate an RGD ID to an Ensembl Transcript ID
*LookupApi* | [**getEnsemblTranscriptMappingUsingPOST**](docs/LookupApi.md#getEnsemblTranscriptMappingUsingPOST) | **POST** /lookup/id/map/EnsemblTranscript | Translate RGD IDs to Ensembl Transcript IDs
*LookupApi* | [**getGTEXMappingUsingGET**](docs/LookupApi.md#getGTEXMappingUsingGET) | **GET** /lookup/id/map/GTEx/{rgdId} | Translate an RGD ID to an GTEx ID
*LookupApi* | [**getGTEXMappingUsingPOST**](docs/LookupApi.md#getGTEXMappingUsingPOST) | **POST** /lookup/id/map/GTEx | Translate RGD IDs to GTEx IDs
*LookupApi* | [**getGenBankNucleotideMappingUsingGET**](docs/LookupApi.md#getGenBankNucleotideMappingUsingGET) | **GET** /lookup/id/map/GenBankNucleotide/{rgdId} | Translate an RGD ID to a GenBank Nucleotide ID
*LookupApi* | [**getGenBankNucleotideMappingUsingPOST**](docs/LookupApi.md#getGenBankNucleotideMappingUsingPOST) | **POST** /lookup/id/map/GenBankNucleotide | Translate RGD IDs to GenBank Nucleotide IDs
*LookupApi* | [**getGenBankProteinMappingUsingGET**](docs/LookupApi.md#getGenBankProteinMappingUsingGET) | **GET** /lookup/id/map/GenBankProtein/{rgdId} | Translate an RGD ID to a GenBank Protein ID
*LookupApi* | [**getGenBankProteinMappingUsingPOST**](docs/LookupApi.md#getGenBankProteinMappingUsingPOST) | **POST** /lookup/id/map/GenBankProtein | Translate RGD IDs to GenBank Protein IDs
*LookupApi* | [**getGeneTypesUsingGET**](docs/LookupApi.md#getGeneTypesUsingGET) | **GET** /lookup/geneTypes | Returns a list of gene types avialable in RGD
*LookupApi* | [**getHGNCMappingUsingGET**](docs/LookupApi.md#getHGNCMappingUsingGET) | **GET** /lookup/id/map/HGNC/{rgdId} | Translate an RGD ID to an HGNC ID
*LookupApi* | [**getHGNCMappingUsingPOST**](docs/LookupApi.md#getHGNCMappingUsingPOST) | **POST** /lookup/id/map/HGNC | Translate RGD IDs to HGNC IDs
*LookupApi* | [**getMGIMappingUsingGET**](docs/LookupApi.md#getMGIMappingUsingGET) | **GET** /lookup/id/map/MGI/{rgdId} | Translate an RGD ID to an MGI ID
*LookupApi* | [**getMGIMappingUsingPOST**](docs/LookupApi.md#getMGIMappingUsingPOST) | **POST** /lookup/id/map/MGI | Translate RGD IDs to MGI IDs
*LookupApi* | [**getMapsUsingGET**](docs/LookupApi.md#getMapsUsingGET) | **GET** /lookup/maps/{speciesTypeKey} | Return a list assembly maps for a species
*LookupApi* | [**getMapsUsingGET1**](docs/LookupApi.md#getMapsUsingGET1) | **GET** /lookup/standardUnit/{accId} | Return a standardUnit for an ontology if it exists
*LookupApi* | [**getNCBIGeneMappingUsingGET**](docs/LookupApi.md#getNCBIGeneMappingUsingGET) | **GET** /lookup/id/map/NCBIGene/{rgdId} | Translate an RGD ID to an NCBI Gene ID
*LookupApi* | [**getNCBIGeneMappingUsingPOST**](docs/LookupApi.md#getNCBIGeneMappingUsingPOST) | **POST** /lookup/id/map/NCBIGene | Translate RGD IDs to NCBI Gene IDs
*LookupApi* | [**getSpeciesTypesUsingGET**](docs/LookupApi.md#getSpeciesTypesUsingGET) | **GET** /lookup/speciesTypeKeys | Return a Map of species type keys available in RGD
*LookupApi* | [**getUniProtMappingUsingGET**](docs/LookupApi.md#getUniProtMappingUsingGET) | **GET** /lookup/id/map/UniProt/{rgdId} | Translate an RGD ID to a UniProt ID
*LookupApi* | [**getUniProtMappingUsingPOST**](docs/LookupApi.md#getUniProtMappingUsingPOST) | **POST** /lookup/id/map/UniProt | Translate RGD IDs to UniProt IDs
*MapApi* | [**getChromosomeByAssemblyUsingGET_0**](docs/MapApi.md#getChromosomeByAssemblyUsingGET_0) | **GET** /maps/chr/{chromosome}/{mapKey} | Return a list of chromosomes
*MapApi* | [**getChromosomesByAssemblyUsingGET_0**](docs/MapApi.md#getChromosomesByAssemblyUsingGET_0) | **GET** /maps/chr/{mapKey} | Return a list of chromosomes
*MapApi* | [**getMapsBySpeciesUsingGET**](docs/MapApi.md#getMapsBySpeciesUsingGET) | **GET** /maps/{speciesTypeKey} | Return a list of assemblies
*OntologyApi* | [**getOntDagsUsingGET**](docs/OntologyApi.md#getOntDagsUsingGET) | **GET** /ontology/ont/{accId} | Returns child and parent terms for Accession ID
*OntologyApi* | [**getTermUsingGET**](docs/OntologyApi.md#getTermUsingGET) | **GET** /ontology/term/{accId} | Returns term for Accession ID
*OntologyApi* | [**isDescendantOfUsingGET**](docs/OntologyApi.md#isDescendantOfUsingGET) | **GET** /ontology/term/{accId1}/{accId2} | Returns true or false for terms
*PathwayApi* | [**getPathwaysWithDiagramsForCategoryUsingGET**](docs/PathwayApi.md#getPathwaysWithDiagramsForCategoryUsingGET) | **GET** /pathways/diagramsForCategory/{category} | Return a list of pathways based on category provided
*PathwayApi* | [**searchPathwaysUsingGET**](docs/PathwayApi.md#searchPathwaysUsingGET) | **GET** /pathways/diagrams/search/{searchString} | Return a list of pathways based on search term
*QtlApi* | [**getMappedQTLByPositionUsingGET**](docs/QtlApi.md#getMappedQTLByPositionUsingGET) | **GET** /qtls/mapped/{chr}/{start}/{stop}/{mapKey} | Returns a list QTL for given position and assembly map
*QtlApi* | [**getQTLByRgdIdUsingGET**](docs/QtlApi.md#getQTLByRgdIdUsingGET) | **GET** /qtls/{rgdId} | Return a QTL for provided RGD ID
*QtlApi* | [**getQtlListByPositionUsingGET**](docs/QtlApi.md#getQtlListByPositionUsingGET) | **GET** /qtls/{chr}/{start}/{stop}/{mapKey} | Returns a list QTL for given position and assembly map
*QuantitativePhenotypeApi* | [**getChartInfoUsingGET**](docs/QuantitativePhenotypeApi.md#getChartInfoUsingGET) | **GET** /phenotype/phenominer/chart/{speciesTypeKey}/{refRgdId}/{termString} | Return a list of quantitative phenotypes values based on a combination of Clinical Measurement, Experimental Condition, Rat Strain, and/or Measurement Method ontology terms.  Results will be all records that match all terms submitted.  Ontology ids should be submitted as a comma delimited list (ex. RS:0000029,CMO:0000155,CMO:0000139).  Species type is an integer value (3&#x3D;rat, 4&#x3D;chinchilla).  Reference RGD ID for a study works like a filter.
*QuantitativePhenotypeApi* | [**getChartInfoUsingGET1**](docs/QuantitativePhenotypeApi.md#getChartInfoUsingGET1) | **GET** /phenotype/phenominer/chart/{speciesTypeKey}/{termString} | Return a list of quantitative phenotypes values based on a combination of Clinical Measurement, Experimental Condition, Rat Strain, and/or Measurement Method ontology terms.  Results will be all records that match all terms submitted.  Ontology ids should be submitted as a comma delimited list (ex. RS:0000029,CMO:0000155,CMO:0000139).  Species type is an integer value (3&#x3D;rat, 4&#x3D;chinchilla)
*RatStrainApi* | [**getAllStrainsUsingGET**](docs/RatStrainApi.md#getAllStrainsUsingGET) | **GET** /strains/all | Return all active strains in RGD
*RatStrainApi* | [**getStrainByRgdIdUsingGET**](docs/RatStrainApi.md#getStrainByRgdIdUsingGET) | **GET** /strains/{rgdId} | Return a strain by RGD ID
*RatStrainApi* | [**getStrainsByPositionUsingGET**](docs/RatStrainApi.md#getStrainsByPositionUsingGET) | **GET** /strains/{chr}/{start}/{stop}/{mapKey} | Return all active strains by position
*SslpApi* | [**getMappedSSLPByPositionUsingGET**](docs/SslpApi.md#getMappedSSLPByPositionUsingGET) | **GET** /sslps/mapped/{chr}/{start}/{stop}/{mapKey} | Returns a list SSLP for given position and assembly map
*StatisticsApi* | [**getActiveObjectCountUsingGET**](docs/StatisticsApi.md#getActiveObjectCountUsingGET) | **GET** /stats/count/activeObject/{speciesTypeKey}/{dateYYYYMMDD} | Count of active objects by type, for specified species and date
*StatisticsApi* | [**getActiveObjectDiffUsingGET**](docs/StatisticsApi.md#getActiveObjectDiffUsingGET) | **GET** /stats/diff/activeObject/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of active objects, by type, for specified species and date range
*StatisticsApi* | [**getGeneTypeCountUsingGET**](docs/StatisticsApi.md#getGeneTypeCountUsingGET) | **GET** /stats/count/geneType/{speciesTypeKey}/{dateYYYYMMDD} | Count of gene types, for specified species and date
*StatisticsApi* | [**getGeneTypeDiffUsingGET**](docs/StatisticsApi.md#getGeneTypeDiffUsingGET) | **GET** /stats/diff/geneType/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of gene types, for specified species and date range
*StatisticsApi* | [**getObjectStatusCountUsingGET**](docs/StatisticsApi.md#getObjectStatusCountUsingGET) | **GET** /stats/count/objectStatus/{speciesTypeKey}/{dateYYYYMMDD} | Count of objects with given status, for specified species and date
*StatisticsApi* | [**getObjectStatusDiffUsingGET**](docs/StatisticsApi.md#getObjectStatusDiffUsingGET) | **GET** /stats/diff/objectStatus/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of objects with given status, for specified species and date range
*StatisticsApi* | [**getObjectsWithRefSeqCountUsingGET**](docs/StatisticsApi.md#getObjectsWithRefSeqCountUsingGET) | **GET** /stats/count/objectWithRefSeq/{speciesTypeKey}/{dateYYYYMMDD} | Count of objects with reference sequence(s), by object type, for specified species and date
*StatisticsApi* | [**getObjectsWithRefSeqDiffUsingGET**](docs/StatisticsApi.md#getObjectsWithRefSeqDiffUsingGET) | **GET** /stats/diff/objectWithRefSeq/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of objects with reference sequence(s), by object type, for specified species and date range
*StatisticsApi* | [**getObjectsWithReferenceCountUsingGET**](docs/StatisticsApi.md#getObjectsWithReferenceCountUsingGET) | **GET** /stats/count/objectWithReference/{speciesTypeKey}/{dateYYYYMMDD} | Count of objects with reference, by object type, for specified species and date
*StatisticsApi* | [**getObjectsWithReferenceDiffUsingGET**](docs/StatisticsApi.md#getObjectsWithReferenceDiffUsingGET) | **GET** /stats/diff/objectWithReference/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of objects with reference, by object type, for specified species and date range
*StatisticsApi* | [**getObjectsWithXDBsCountUsingGET**](docs/StatisticsApi.md#getObjectsWithXDBsCountUsingGET) | **GET** /stats/count/objectWithXdb/{speciesTypeKey}/{objectKey}/{dateYYYYMMDD} | Count of objects with external database ids, by database id, for specified species, object type and date
*StatisticsApi* | [**getObjectsWithXDBsDiffUsingGET**](docs/StatisticsApi.md#getObjectsWithXDBsDiffUsingGET) | **GET** /stats/diff/objectWithXdb/{speciesTypeKey}/{objectKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of objects with external database ids, by database id, for specified species, object type and date range
*StatisticsApi* | [**getProteinInteractionCountUsingGET**](docs/StatisticsApi.md#getProteinInteractionCountUsingGET) | **GET** /stats/count/proteinInteraction/{speciesTypeKey}/{dateYYYYMMDD} | Count of protein interactions, for specified species and date
*StatisticsApi* | [**getProteinInteractionDiffUsingGET**](docs/StatisticsApi.md#getProteinInteractionDiffUsingGET) | **GET** /stats/diff/proteinInteraction/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of protein interactions, for specified species and date range
*StatisticsApi* | [**getQtlInheritanceTypeCountUsingGET**](docs/StatisticsApi.md#getQtlInheritanceTypeCountUsingGET) | **GET** /stats/count/qtlInheritanceType/{speciesTypeKey}/{dateYYYYMMDD} | Count of strains, by qtl inheritance type, for specified species and date
*StatisticsApi* | [**getQtlInheritanceTypeDiffUsingGET**](docs/StatisticsApi.md#getQtlInheritanceTypeDiffUsingGET) | **GET** /stats/diff/qtlInheritanceType/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of strains, by qtl inheritance type, for specified species and date range
*StatisticsApi* | [**getRetiredObjectCountUsingGET**](docs/StatisticsApi.md#getRetiredObjectCountUsingGET) | **GET** /stats/count/retiredObject/{speciesTypeKey}/{dateYYYYMMDD} | Count of retired objects by type, for specified species and date
*StatisticsApi* | [**getRetiredObjectDiffUsingGET**](docs/StatisticsApi.md#getRetiredObjectDiffUsingGET) | **GET** /stats/diff/retiredObject/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of retired objects, by type, for specified species and date range
*StatisticsApi* | [**getStrainTypeCountUsingGET**](docs/StatisticsApi.md#getStrainTypeCountUsingGET) | **GET** /stats/count/strainType/{speciesTypeKey}/{dateYYYYMMDD} | Count of strain types, for specified species and date
*StatisticsApi* | [**getStrainTypeDiffUsingGET**](docs/StatisticsApi.md#getStrainTypeDiffUsingGET) | **GET** /stats/diff/strainType/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of strain types, for specified species and date range
*StatisticsApi* | [**getTermStatsUsingGET**](docs/StatisticsApi.md#getTermStatsUsingGET) | **GET** /stats/term/{accId}/{filterAccId} | getTermStats
*StatisticsApi* | [**getWithdrawnObjectCountUsingGET**](docs/StatisticsApi.md#getWithdrawnObjectCountUsingGET) | **GET** /stats/count/withdrawnObject/{speciesTypeKey}/{dateYYYYMMDD} | Count of withdrawn objects by type, for specified species and date
*StatisticsApi* | [**getWithdrawnObjectDiffUsingGET**](docs/StatisticsApi.md#getWithdrawnObjectDiffUsingGET) | **GET** /stats/diff/withdrawnObject/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of withdrawn objects, by type, for specified species and date range
*StatisticsApi* | [**getXdbsCountUsingGET**](docs/StatisticsApi.md#getXdbsCountUsingGET) | **GET** /stats/count/xdb/{speciesTypeKey}/{dateYYYYMMDD} | Count of external database ids, for specied species and date
*StatisticsApi* | [**getXdbsDiffUsingGET**](docs/StatisticsApi.md#getXdbsDiffUsingGET) | **GET** /stats/diff/xdb/{speciesTypeKey}/{dateFromYYYYMMDD}/{dateToYYYYMMDD} | Count difference of external database ids, for specified species and date range


## Documentation for Models

 - [AnnotatedGeneRequest](docs/AnnotatedGeneRequest.md)
 - [Annotation](docs/Annotation.md)
 - [AnnotationRequest](docs/AnnotationRequest.md)
 - [Chromosome](docs/Chromosome.md)
 - [EnrichmentGeneRequest](docs/EnrichmentGeneRequest.md)
 - [EnrichmentRequest](docs/EnrichmentRequest.md)
 - [Gene](docs/Gene.md)
 - [Map](docs/Map.md)
 - [MapPair](docs/MapPair.md)
 - [MappedGene](docs/MappedGene.md)
 - [MappedGenePosition](docs/MappedGenePosition.md)
 - [MappedQTL](docs/MappedQTL.md)
 - [MappedSSLP](docs/MappedSSLP.md)
 - [OrthologRequest](docs/OrthologRequest.md)
 - [Pathway](docs/Pathway.md)
 - [PathwayObject](docs/PathwayObject.md)
 - [QTL](docs/QTL.md)
 - [RGDIDListRequest](docs/RGDIDListRequest.md)
 - [Reference](docs/Reference.md)
 - [SSLP](docs/SSLP.md)
 - [Status](docs/Status.md)
 - [Strain](docs/Strain.md)
 - [Term](docs/Term.md)
 - [TermXRef](docs/TermXRef.md)


<a id="documentation-for-authorization"></a>
## Documentation for Authorization

Endpoints do not require authorization.


## Recommendation

It's recommended to create an instance of `ApiClient` per thread in a multithreaded environment to avoid any potential issues.

## Author

RGD.Data2@mcw.edu

