/*
 * Enterobase-API
 *  API for EnteroBase (https://enterobase.warwick.ac.uk)   EnteroBase is a user-friendly online resource, where users can upload their  own sequencing data for de novo assembly by a stream-lined pipeline. The assemblies  are used for calling MLST and wgMLST patterns, allowing users to compare their strains  to publically available genotyping data from other EnteroBase users, GenBank and classical MLST databases.  Click here to find how to get and use an API token: http://bit.ly/1TKlaOU 
 *
 * The version of the OpenAPI document: v2.0
 * Contact: enterobase@warwick.ac.uk
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package org.openapitools.client.api;

import org.openapitools.client.ApiCallback;
import org.openapitools.client.ApiClient;
import org.openapitools.client.ApiException;
import org.openapitools.client.ApiResponse;
import org.openapitools.client.Configuration;
import org.openapitools.client.Pair;
import org.openapitools.client.ProgressRequestBody;
import org.openapitools.client.ProgressResponseBody;

import com.google.gson.reflect.TypeToken;

import java.io.IOException;



import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class AllelesApi {
    private ApiClient localVarApiClient;
    private int localHostIndex;
    private String localCustomBaseUrl;

    public AllelesApi() {
        this(Configuration.getDefaultApiClient());
    }

    public AllelesApi(ApiClient apiClient) {
        this.localVarApiClient = apiClient;
    }

    public ApiClient getApiClient() {
        return localVarApiClient;
    }

    public void setApiClient(ApiClient apiClient) {
        this.localVarApiClient = apiClient;
    }

    public int getHostIndex() {
        return localHostIndex;
    }

    public void setHostIndex(int hostIndex) {
        this.localHostIndex = hostIndex;
    }

    public String getCustomBaseUrl() {
        return localCustomBaseUrl;
    }

    public void setCustomBaseUrl(String customBaseUrl) {
        this.localCustomBaseUrl = customBaseUrl;
    }

    /**
     * Build call for apiV20DatabaseSchemeAllelesGet
     * @param locus  (required)
     * @param database Species database name (senterica, ecoli, yersinia, mcatarrhalis) for Salmonella, Escherichia, Yersinia, Moraxella respectively (required)
     * @param scheme  (required)
     * @param barcode Unique barcode for Strain records, &lt;database prefix&gt;_&lt;ID code&gt; e.g. SAL_AA0001AA (optional)
     * @param offset  (optional, default to 0)
     * @param alleleId  (optional)
     * @param onlyFields  (optional)
     * @param seq  (optional)
     * @param reldate  (optional)
     * @param limit  (optional, default to 50)
     * @param _callback Callback for upload/download progress
     * @return Call to execute
     * @throws ApiException If fail to serialize the request body object
     * @http.response.details
     <table summary="Response Details" border="1">
        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
        <tr><td> 200 </td><td> List of alleles objects </td><td>  -  </td></tr>
        <tr><td> 400 </td><td> Malformed request, contains an error </td><td>  -  </td></tr>
        <tr><td> 403 </td><td> Unauthorised access for this specific resource </td><td>  -  </td></tr>
     </table>
     */
    public okhttp3.Call apiV20DatabaseSchemeAllelesGetCall(String locus, String database, String scheme, List<String> barcode, Integer offset, String alleleId, List<String> onlyFields, String seq, Integer reldate, Integer limit, final ApiCallback _callback) throws ApiException {
        String basePath = null;
        // Operation Servers
        String[] localBasePaths = new String[] {  };

        // Determine Base Path to Use
        if (localCustomBaseUrl != null){
            basePath = localCustomBaseUrl;
        } else if ( localBasePaths.length > 0 ) {
            basePath = localBasePaths[localHostIndex];
        } else {
            basePath = null;
        }

        Object localVarPostBody = null;

        // create path and map variables
        String localVarPath = "/api/v2.0/{database}/{scheme}/alleles"
            .replace("{" + "database" + "}", localVarApiClient.escapeString(database.toString()))
            .replace("{" + "scheme" + "}", localVarApiClient.escapeString(scheme.toString()));

        List<Pair> localVarQueryParams = new ArrayList<Pair>();
        List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();
        Map<String, String> localVarHeaderParams = new HashMap<String, String>();
        Map<String, String> localVarCookieParams = new HashMap<String, String>();
        Map<String, Object> localVarFormParams = new HashMap<String, Object>();

        if (barcode != null) {
            localVarCollectionQueryParams.addAll(localVarApiClient.parameterToPairs("multi", "barcode", barcode));
        }

        if (locus != null) {
            localVarQueryParams.addAll(localVarApiClient.parameterToPair("locus", locus));
        }

        if (offset != null) {
            localVarQueryParams.addAll(localVarApiClient.parameterToPair("offset", offset));
        }

        if (alleleId != null) {
            localVarQueryParams.addAll(localVarApiClient.parameterToPair("allele_id", alleleId));
        }

        if (onlyFields != null) {
            localVarCollectionQueryParams.addAll(localVarApiClient.parameterToPairs("multi", "only_fields", onlyFields));
        }

        if (seq != null) {
            localVarQueryParams.addAll(localVarApiClient.parameterToPair("seq", seq));
        }

        if (reldate != null) {
            localVarQueryParams.addAll(localVarApiClient.parameterToPair("reldate", reldate));
        }

        if (limit != null) {
            localVarQueryParams.addAll(localVarApiClient.parameterToPair("limit", limit));
        }

        final String[] localVarAccepts = {
        };
        final String localVarAccept = localVarApiClient.selectHeaderAccept(localVarAccepts);
        if (localVarAccept != null) {
            localVarHeaderParams.put("Accept", localVarAccept);
        }

        final String[] localVarContentTypes = {
        };
        final String localVarContentType = localVarApiClient.selectHeaderContentType(localVarContentTypes);
        if (localVarContentType != null) {
            localVarHeaderParams.put("Content-Type", localVarContentType);
        }

        String[] localVarAuthNames = new String[] {  };
        return localVarApiClient.buildCall(basePath, localVarPath, "GET", localVarQueryParams, localVarCollectionQueryParams, localVarPostBody, localVarHeaderParams, localVarCookieParams, localVarFormParams, localVarAuthNames, _callback);
    }

    @SuppressWarnings("rawtypes")
    private okhttp3.Call apiV20DatabaseSchemeAllelesGetValidateBeforeCall(String locus, String database, String scheme, List<String> barcode, Integer offset, String alleleId, List<String> onlyFields, String seq, Integer reldate, Integer limit, final ApiCallback _callback) throws ApiException {
        // verify the required parameter 'locus' is set
        if (locus == null) {
            throw new ApiException("Missing the required parameter 'locus' when calling apiV20DatabaseSchemeAllelesGet(Async)");
        }

        // verify the required parameter 'database' is set
        if (database == null) {
            throw new ApiException("Missing the required parameter 'database' when calling apiV20DatabaseSchemeAllelesGet(Async)");
        }

        // verify the required parameter 'scheme' is set
        if (scheme == null) {
            throw new ApiException("Missing the required parameter 'scheme' when calling apiV20DatabaseSchemeAllelesGet(Async)");
        }

        return apiV20DatabaseSchemeAllelesGetCall(locus, database, scheme, barcode, offset, alleleId, onlyFields, seq, reldate, limit, _callback);

    }

    /**
     * 
     * Alleles  data 
     * @param locus  (required)
     * @param database Species database name (senterica, ecoli, yersinia, mcatarrhalis) for Salmonella, Escherichia, Yersinia, Moraxella respectively (required)
     * @param scheme  (required)
     * @param barcode Unique barcode for Strain records, &lt;database prefix&gt;_&lt;ID code&gt; e.g. SAL_AA0001AA (optional)
     * @param offset  (optional, default to 0)
     * @param alleleId  (optional)
     * @param onlyFields  (optional)
     * @param seq  (optional)
     * @param reldate  (optional)
     * @param limit  (optional, default to 50)
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     * @http.response.details
     <table summary="Response Details" border="1">
        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
        <tr><td> 200 </td><td> List of alleles objects </td><td>  -  </td></tr>
        <tr><td> 400 </td><td> Malformed request, contains an error </td><td>  -  </td></tr>
        <tr><td> 403 </td><td> Unauthorised access for this specific resource </td><td>  -  </td></tr>
     </table>
     */
    public void apiV20DatabaseSchemeAllelesGet(String locus, String database, String scheme, List<String> barcode, Integer offset, String alleleId, List<String> onlyFields, String seq, Integer reldate, Integer limit) throws ApiException {
        apiV20DatabaseSchemeAllelesGetWithHttpInfo(locus, database, scheme, barcode, offset, alleleId, onlyFields, seq, reldate, limit);
    }

    /**
     * 
     * Alleles  data 
     * @param locus  (required)
     * @param database Species database name (senterica, ecoli, yersinia, mcatarrhalis) for Salmonella, Escherichia, Yersinia, Moraxella respectively (required)
     * @param scheme  (required)
     * @param barcode Unique barcode for Strain records, &lt;database prefix&gt;_&lt;ID code&gt; e.g. SAL_AA0001AA (optional)
     * @param offset  (optional, default to 0)
     * @param alleleId  (optional)
     * @param onlyFields  (optional)
     * @param seq  (optional)
     * @param reldate  (optional)
     * @param limit  (optional, default to 50)
     * @return ApiResponse&lt;Void&gt;
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     * @http.response.details
     <table summary="Response Details" border="1">
        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
        <tr><td> 200 </td><td> List of alleles objects </td><td>  -  </td></tr>
        <tr><td> 400 </td><td> Malformed request, contains an error </td><td>  -  </td></tr>
        <tr><td> 403 </td><td> Unauthorised access for this specific resource </td><td>  -  </td></tr>
     </table>
     */
    public ApiResponse<Void> apiV20DatabaseSchemeAllelesGetWithHttpInfo(String locus, String database, String scheme, List<String> barcode, Integer offset, String alleleId, List<String> onlyFields, String seq, Integer reldate, Integer limit) throws ApiException {
        okhttp3.Call localVarCall = apiV20DatabaseSchemeAllelesGetValidateBeforeCall(locus, database, scheme, barcode, offset, alleleId, onlyFields, seq, reldate, limit, null);
        return localVarApiClient.execute(localVarCall);
    }

    /**
     *  (asynchronously)
     * Alleles  data 
     * @param locus  (required)
     * @param database Species database name (senterica, ecoli, yersinia, mcatarrhalis) for Salmonella, Escherichia, Yersinia, Moraxella respectively (required)
     * @param scheme  (required)
     * @param barcode Unique barcode for Strain records, &lt;database prefix&gt;_&lt;ID code&gt; e.g. SAL_AA0001AA (optional)
     * @param offset  (optional, default to 0)
     * @param alleleId  (optional)
     * @param onlyFields  (optional)
     * @param seq  (optional)
     * @param reldate  (optional)
     * @param limit  (optional, default to 50)
     * @param _callback The callback to be executed when the API call finishes
     * @return The request call
     * @throws ApiException If fail to process the API call, e.g. serializing the request body object
     * @http.response.details
     <table summary="Response Details" border="1">
        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
        <tr><td> 200 </td><td> List of alleles objects </td><td>  -  </td></tr>
        <tr><td> 400 </td><td> Malformed request, contains an error </td><td>  -  </td></tr>
        <tr><td> 403 </td><td> Unauthorised access for this specific resource </td><td>  -  </td></tr>
     </table>
     */
    public okhttp3.Call apiV20DatabaseSchemeAllelesGetAsync(String locus, String database, String scheme, List<String> barcode, Integer offset, String alleleId, List<String> onlyFields, String seq, Integer reldate, Integer limit, final ApiCallback<Void> _callback) throws ApiException {

        okhttp3.Call localVarCall = apiV20DatabaseSchemeAllelesGetValidateBeforeCall(locus, database, scheme, barcode, offset, alleleId, onlyFields, seq, reldate, limit, _callback);
        localVarApiClient.executeAsync(localVarCall, _callback);
        return localVarCall;
    }
}
