# openapi-java-client

BioLink API
- API version: 1.1.14
  - Build date: 2024-10-12T12:41:42.559774-04:00[America/New_York]
  - Generator version: 7.9.0

API integration layer for linked biological objects.

 __Source:__ https://github.com/biolink/biolink-api/


*Automatically generated by the [OpenAPI Generator](https://openapi-generator.tech)*


## Requirements

Building the API client library requires:
1. Java 1.8+
2. Maven (3.8.3+)/Gradle (7.2+)

## Installation

To install the API client library to your local Maven repository, simply execute:

```shell
mvn clean install
```

To deploy it to a remote Maven repository instead, configure the settings of the repository and execute:

```shell
mvn clean deploy
```

Refer to the [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html) for more information.

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>org.openapitools</groupId>
  <artifactId>openapi-java-client</artifactId>
  <version>1.1.14</version>
  <scope>compile</scope>
</dependency>
```

### Gradle users

Add this dependency to your project's build file:

```groovy
  repositories {
    mavenCentral()     // Needed if the 'openapi-java-client' jar has been published to maven central.
    mavenLocal()       // Needed if the 'openapi-java-client' jar has been published to the local maven repo.
  }

  dependencies {
     implementation "org.openapitools:openapi-java-client:1.1.14"
  }
```

### Others

At first generate the JAR by executing:

```shell
mvn clean package
```

Then manually install the following JARs:

* `target/openapi-java-client-1.1.14.jar`
* `target/lib/*.jar`

## Getting Started

Please follow the [installation](#installation) instruction and execute the following Java code:

```java

// Import classes:
import org.openapitools.client.ApiClient;
import org.openapitools.client.ApiException;
import org.openapitools.client.Configuration;
import org.openapitools.client.model.*;
import org.openapitools.client.api.AssociationApi;

public class Example {
  public static void main(String[] args) {
    ApiClient defaultClient = Configuration.getDefaultApiClient();
    defaultClient.setBasePath("/api");

    AssociationApi apiInstance = new AssociationApi(defaultClient);
    String associationType = "associationType_example"; // String | Association type, eg gene_phenotype
    Integer rows = 100; // Integer | number of rows
    Integer start = 56; // Integer | beginning row
    String evidence = "evidence_example"; // String | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2
    Boolean unselectEvidence = false; // Boolean | If true, excludes evidence objects in response
    Boolean excludeAutomaticAssertions = false; // Boolean | If true, excludes associations that involve IEAs (ECO:0000501)
    Boolean useCompactAssociations = false; // Boolean | If true, returns results in compact associations format
    String subject = "subject_example"; // String | Subject CURIE
    String _object = "_object_example"; // String | Object CURIE
    try {
      List<AssociationResults> result = apiInstance.getAssociationBySubjectAndAssocType(associationType, rows, start, evidence, unselectEvidence, excludeAutomaticAssertions, useCompactAssociations, subject, _object);
      System.out.println(result);
    } catch (ApiException e) {
      System.err.println("Exception when calling AssociationApi#getAssociationBySubjectAndAssocType");
      System.err.println("Status code: " + e.getCode());
      System.err.println("Reason: " + e.getResponseBody());
      System.err.println("Response headers: " + e.getResponseHeaders());
      e.printStackTrace();
    }
  }
}

```

## Documentation for API Endpoints

All URIs are relative to */api*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AssociationApi* | [**getAssociationBySubjectAndAssocType**](docs/AssociationApi.md#getAssociationBySubjectAndAssocType) | **GET** /association/type/{association_type} | Returns list of matching associations of a given type
*AssociationApi* | [**getAssociationBySubjectAndObjectCategorySearch**](docs/AssociationApi.md#getAssociationBySubjectAndObjectCategorySearch) | **GET** /association/find/{subject_category}/{object_category} | Returns list of matching associations between a given subject and object category
*AssociationApi* | [**getAssociationBySubjectCategorySearch**](docs/AssociationApi.md#getAssociationBySubjectCategorySearch) | **GET** /association/find/{subject_category} | Returns list of matching associations for a given subject category
*AssociationApi* | [**getAssociationObject**](docs/AssociationApi.md#getAssociationObject) | **GET** /association/{id} | Returns the association with a given identifier
*AssociationApi* | [**getAssociationsBetween**](docs/AssociationApi.md#getAssociationsBetween) | **GET** /association/between/{subject}/{object} | Returns associations connecting two entities
*AssociationApi* | [**getAssociationsFrom**](docs/AssociationApi.md#getAssociationsFrom) | **GET** /association/from/{subject} | Returns list of matching associations starting from a given subject (source)
*AssociationApi* | [**getAssociationsTo**](docs/AssociationApi.md#getAssociationsTo) | **GET** /association/to/{object} | Returns list of matching associations pointing to a given object (target)
*BioentityApi* | [**getAnatomyGeneAssociations**](docs/BioentityApi.md#getAnatomyGeneAssociations) | **GET** /bioentity/anatomy/{id}/genes | Returns genes associated with a given anatomy
*BioentityApi* | [**getAnatomyGeneByTaxonAssociations**](docs/BioentityApi.md#getAnatomyGeneByTaxonAssociations) | **GET** /bioentity/anatomy/{id}/genes/{taxid} | Returns gene IDs for all genes associated with a given anatomy, filtered by taxon
*BioentityApi* | [**getCaseDiseaseAssociations**](docs/BioentityApi.md#getCaseDiseaseAssociations) | **GET** /bioentity/case/{id}/diseases | Returns diseases associated with a case
*BioentityApi* | [**getCaseGenotypeAssociations**](docs/BioentityApi.md#getCaseGenotypeAssociations) | **GET** /bioentity/case/{id}/genotypes | Returns genotypes associated with a case
*BioentityApi* | [**getCaseModelAssociations**](docs/BioentityApi.md#getCaseModelAssociations) | **GET** /bioentity/case/{id}/models | Returns models associated with a case
*BioentityApi* | [**getCasePhenotypeAssociations**](docs/BioentityApi.md#getCasePhenotypeAssociations) | **GET** /bioentity/case/{id}/phenotypes | Returns phenotypes associated with a case
*BioentityApi* | [**getCaseVariantAssociations**](docs/BioentityApi.md#getCaseVariantAssociations) | **GET** /bioentity/case/{id}/variants | Returns variants associated with a case
*BioentityApi* | [**getDiseaseCaseAssociations**](docs/BioentityApi.md#getDiseaseCaseAssociations) | **GET** /bioentity/disease/{id}/cases | Returns cases associated with a disease
*BioentityApi* | [**getDiseaseGeneAssociations**](docs/BioentityApi.md#getDiseaseGeneAssociations) | **GET** /bioentity/disease/{id}/genes | Returns genes associated with a disease
*BioentityApi* | [**getDiseaseGenotypeAssociations**](docs/BioentityApi.md#getDiseaseGenotypeAssociations) | **GET** /bioentity/disease/{id}/genotypes | Returns genotypes associated with a disease
*BioentityApi* | [**getDiseaseModelAssociations**](docs/BioentityApi.md#getDiseaseModelAssociations) | **GET** /bioentity/disease/{id}/models | Returns associations to models of the disease
*BioentityApi* | [**getDiseaseModelTaxonAssociations**](docs/BioentityApi.md#getDiseaseModelTaxonAssociations) | **GET** /bioentity/disease/{id}/models/{taxon} | Returns associations to models of the disease constrained by taxon
*BioentityApi* | [**getDiseasePathwayAssociations**](docs/BioentityApi.md#getDiseasePathwayAssociations) | **GET** /bioentity/disease/{id}/pathways | Returns pathways associated with a disease
*BioentityApi* | [**getDiseasePhenotypeAssociations**](docs/BioentityApi.md#getDiseasePhenotypeAssociations) | **GET** /bioentity/disease/{id}/phenotypes | Returns phenotypes associated with disease
*BioentityApi* | [**getDiseasePublicationAssociations**](docs/BioentityApi.md#getDiseasePublicationAssociations) | **GET** /bioentity/disease/{id}/publications | Returns publications associated with a disease
*BioentityApi* | [**getDiseaseSubstanceAssociations**](docs/BioentityApi.md#getDiseaseSubstanceAssociations) | **GET** /bioentity/disease/{id}/treatment | Returns substances associated with a disease
*BioentityApi* | [**getDiseaseVariantAssociations**](docs/BioentityApi.md#getDiseaseVariantAssociations) | **GET** /bioentity/disease/{id}/variants | Returns variants associated with a disease
*BioentityApi* | [**getFunctionAssociations**](docs/BioentityApi.md#getFunctionAssociations) | **GET** /bioentity/function/{id} | Returns annotations associated to a function term
*BioentityApi* | [**getFunctionGeneAssociations**](docs/BioentityApi.md#getFunctionGeneAssociations) | **GET** /bioentity/function/{id}/genes | Returns genes associated to a GO term
*BioentityApi* | [**getFunctionPublicationAssociations**](docs/BioentityApi.md#getFunctionPublicationAssociations) | **GET** /bioentity/function/{id}/publications | Returns publications associated to a GO term
*BioentityApi* | [**getFunctionTaxonAssociations**](docs/BioentityApi.md#getFunctionTaxonAssociations) | **GET** /bioentity/function/{id}/taxons | Returns taxons associated to a GO term
*BioentityApi* | [**getGeneAnatomyAssociations**](docs/BioentityApi.md#getGeneAnatomyAssociations) | **GET** /bioentity/gene/{id}/anatomy | Returns anatomical entities associated with a gene
*BioentityApi* | [**getGeneCaseAssociations**](docs/BioentityApi.md#getGeneCaseAssociations) | **GET** /bioentity/gene/{id}/cases | Returns cases associated with a gene
*BioentityApi* | [**getGeneDiseaseAssociations**](docs/BioentityApi.md#getGeneDiseaseAssociations) | **GET** /bioentity/gene/{id}/diseases | Returns diseases associated with gene
*BioentityApi* | [**getGeneExpressionAssociations**](docs/BioentityApi.md#getGeneExpressionAssociations) | **GET** /bioentity/gene/{id}/expression/anatomy | Returns expression events for a gene
*BioentityApi* | [**getGeneFunctionAssociations**](docs/BioentityApi.md#getGeneFunctionAssociations) | **GET** /bioentity/gene/{id}/function | Returns function associations for a gene
*BioentityApi* | [**getGeneGenotypeAssociations**](docs/BioentityApi.md#getGeneGenotypeAssociations) | **GET** /bioentity/gene/{id}/genotypes | Returns genotypes associated with a gene
*BioentityApi* | [**getGeneHomologAssociations**](docs/BioentityApi.md#getGeneHomologAssociations) | **GET** /bioentity/gene/{id}/homologs | Returns homologs for a gene
*BioentityApi* | [**getGeneInteractions**](docs/BioentityApi.md#getGeneInteractions) | **GET** /bioentity/gene/{id}/interactions | Returns interactions for a gene
*BioentityApi* | [**getGeneModelAssociations**](docs/BioentityApi.md#getGeneModelAssociations) | **GET** /bioentity/gene/{id}/models | Returns models associated with a gene
*BioentityApi* | [**getGeneOrthologDiseaseAssociations**](docs/BioentityApi.md#getGeneOrthologDiseaseAssociations) | **GET** /bioentity/gene/{id}/ortholog/diseases | Return diseases associated with orthologs of a gene
*BioentityApi* | [**getGeneOrthologPhenotypeAssociations**](docs/BioentityApi.md#getGeneOrthologPhenotypeAssociations) | **GET** /bioentity/gene/{id}/ortholog/phenotypes | Return phenotypes associated with orthologs for a gene
*BioentityApi* | [**getGenePathwayAssociations**](docs/BioentityApi.md#getGenePathwayAssociations) | **GET** /bioentity/gene/{id}/pathways | Returns pathways associated with gene
*BioentityApi* | [**getGenePhenotypeAssociations**](docs/BioentityApi.md#getGenePhenotypeAssociations) | **GET** /bioentity/gene/{id}/phenotypes | Returns phenotypes associated with gene
*BioentityApi* | [**getGenePublicationAssociations**](docs/BioentityApi.md#getGenePublicationAssociations) | **GET** /bioentity/gene/{id}/publications | Returns publications associated with a gene
*BioentityApi* | [**getGeneVariantAssociations**](docs/BioentityApi.md#getGeneVariantAssociations) | **GET** /bioentity/gene/{id}/variants | Returns variants associated with a gene
*BioentityApi* | [**getGenericAssociations**](docs/BioentityApi.md#getGenericAssociations) | **GET** /bioentity/{id}/associations | Returns associations for an entity regardless of the type
*BioentityApi* | [**getGenericObject**](docs/BioentityApi.md#getGenericObject) | **GET** /bioentity/{id} | Returns basic info on object of any type
*BioentityApi* | [**getGenericObjectByType**](docs/BioentityApi.md#getGenericObjectByType) | **GET** /bioentity/{type}/{id} | Return basic info on an object for a given type
*BioentityApi* | [**getGenotypeCaseAssociations**](docs/BioentityApi.md#getGenotypeCaseAssociations) | **GET** /bioentity/genotype/{id}/cases | Returns cases associated with a genotype
*BioentityApi* | [**getGenotypeDiseaseAssociations**](docs/BioentityApi.md#getGenotypeDiseaseAssociations) | **GET** /bioentity/genotype/{id}/diseases | Returns diseases associated with a genotype
*BioentityApi* | [**getGenotypeGeneAssociations**](docs/BioentityApi.md#getGenotypeGeneAssociations) | **GET** /bioentity/genotype/{id}/genes | Returns genes associated with a genotype
*BioentityApi* | [**getGenotypeGenotypeAssociations**](docs/BioentityApi.md#getGenotypeGenotypeAssociations) | **GET** /bioentity/genotype/{id}/genotypes | Returns genotypes-genotype associations
*BioentityApi* | [**getGenotypeModelAssociations**](docs/BioentityApi.md#getGenotypeModelAssociations) | **GET** /bioentity/genotype/{id}/models | Returns models associated with a genotype
*BioentityApi* | [**getGenotypePhenotypeAssociations**](docs/BioentityApi.md#getGenotypePhenotypeAssociations) | **GET** /bioentity/genotype/{id}/phenotypes | Returns phenotypes associated with a genotype
*BioentityApi* | [**getGenotypePublicationAssociations**](docs/BioentityApi.md#getGenotypePublicationAssociations) | **GET** /bioentity/genotype/{id}/publications | Returns publications associated with a genotype
*BioentityApi* | [**getGenotypeVariantAssociations**](docs/BioentityApi.md#getGenotypeVariantAssociations) | **GET** /bioentity/genotype/{id}/variants | Returns genotypes-variant associations
*BioentityApi* | [**getGotermGeneAssociations**](docs/BioentityApi.md#getGotermGeneAssociations) | **GET** /bioentity/goterm/{id}/genes | Returns associations to GO terms for a gene
*BioentityApi* | [**getModelCaseAssociations**](docs/BioentityApi.md#getModelCaseAssociations) | **GET** /bioentity/model/{id}/cases | Returns cases associated with a model
*BioentityApi* | [**getModelDiseaseAssociations**](docs/BioentityApi.md#getModelDiseaseAssociations) | **GET** /bioentity/model/{id}/diseases | Returns diseases associated with a model
*BioentityApi* | [**getModelGeneAssociations**](docs/BioentityApi.md#getModelGeneAssociations) | **GET** /bioentity/model/{id}/genes | Returns genes associated with a model
*BioentityApi* | [**getModelGenotypeAssociations**](docs/BioentityApi.md#getModelGenotypeAssociations) | **GET** /bioentity/model/{id}/genotypes | Returns genotypes associated with a model
*BioentityApi* | [**getModelPhenotypeAssociations**](docs/BioentityApi.md#getModelPhenotypeAssociations) | **GET** /bioentity/model/{id}/phenotypes | Returns phenotypes associated with a model
*BioentityApi* | [**getModelPublicationAssociations**](docs/BioentityApi.md#getModelPublicationAssociations) | **GET** /bioentity/model/{id}/publications | Returns publications associated with a model
*BioentityApi* | [**getModelVariantAssociations**](docs/BioentityApi.md#getModelVariantAssociations) | **GET** /bioentity/model/{id}/variants | Returns variants associated with a model
*BioentityApi* | [**getPathwayDiseaseAssociations**](docs/BioentityApi.md#getPathwayDiseaseAssociations) | **GET** /bioentity/pathway/{id}/diseases | Returns diseases associated with a pathway
*BioentityApi* | [**getPathwayGeneAssociations**](docs/BioentityApi.md#getPathwayGeneAssociations) | **GET** /bioentity/pathway/{id}/genes | Returns genes associated with a pathway
*BioentityApi* | [**getPathwayPhenotypeAssociations**](docs/BioentityApi.md#getPathwayPhenotypeAssociations) | **GET** /bioentity/pathway/{id}/phenotypes | Returns phenotypes associated with a pathway
*BioentityApi* | [**getPhenotypeAnatomyAssociations**](docs/BioentityApi.md#getPhenotypeAnatomyAssociations) | **GET** /bioentity/phenotype/{id}/anatomy | Returns anatomical entities associated with a phenotype
*BioentityApi* | [**getPhenotypeCaseAssociations**](docs/BioentityApi.md#getPhenotypeCaseAssociations) | **GET** /bioentity/phenotype/{id}/cases | Returns cases associated with a phenotype
*BioentityApi* | [**getPhenotypeDiseaseAssociations**](docs/BioentityApi.md#getPhenotypeDiseaseAssociations) | **GET** /bioentity/phenotype/{id}/diseases | Returns diseases associated with a phenotype
*BioentityApi* | [**getPhenotypeGeneAssociations**](docs/BioentityApi.md#getPhenotypeGeneAssociations) | **GET** /bioentity/phenotype/{id}/genes | Returns genes associated with a phenotype
*BioentityApi* | [**getPhenotypeGeneByTaxonAssociations**](docs/BioentityApi.md#getPhenotypeGeneByTaxonAssociations) | **GET** /bioentity/phenotype/{id}/gene/{taxid}/ids | Returns gene IDs for all genes associated with a given phenotype, filtered by taxon
*BioentityApi* | [**getPhenotypeGenotypeAssociations**](docs/BioentityApi.md#getPhenotypeGenotypeAssociations) | **GET** /bioentity/phenotype/{id}/genotypes | Returns genotypes associated with a phenotype
*BioentityApi* | [**getPhenotypePathwayAssociations**](docs/BioentityApi.md#getPhenotypePathwayAssociations) | **GET** /bioentity/phenotype/{id}/pathways | Returns pathways associated with a phenotype
*BioentityApi* | [**getPhenotypePublicationAssociations**](docs/BioentityApi.md#getPhenotypePublicationAssociations) | **GET** /bioentity/phenotype/{id}/publications | Returns publications associated with a phenotype
*BioentityApi* | [**getPhenotypeVariantAssociations**](docs/BioentityApi.md#getPhenotypeVariantAssociations) | **GET** /bioentity/phenotype/{id}/variants | Returns variants associated with a phenotype
*BioentityApi* | [**getPublicationDiseaseAssociations**](docs/BioentityApi.md#getPublicationDiseaseAssociations) | **GET** /bioentity/publication/{id}/diseases | Returns diseases associated with a publication
*BioentityApi* | [**getPublicationGeneAssociations**](docs/BioentityApi.md#getPublicationGeneAssociations) | **GET** /bioentity/publication/{id}/genes | Returns genes associated with a publication
*BioentityApi* | [**getPublicationGenotypeAssociations**](docs/BioentityApi.md#getPublicationGenotypeAssociations) | **GET** /bioentity/publication/{id}/genotypes | Returns genotypes associated with a publication
*BioentityApi* | [**getPublicationModelAssociations**](docs/BioentityApi.md#getPublicationModelAssociations) | **GET** /bioentity/publication/{id}/models | Returns models associated with a publication
*BioentityApi* | [**getPublicationPhenotypeAssociations**](docs/BioentityApi.md#getPublicationPhenotypeAssociations) | **GET** /bioentity/publication/{id}/phenotypes | Returns phenotypes associated with a publication
*BioentityApi* | [**getPublicationVariantAssociations**](docs/BioentityApi.md#getPublicationVariantAssociations) | **GET** /bioentity/publication/{id}/variants | Returns variants associated with a publication
*BioentityApi* | [**getSubstanceParticipantInAssociations**](docs/BioentityApi.md#getSubstanceParticipantInAssociations) | **GET** /bioentity/substance/{id}/participant_in | Returns associations between an activity and process and the specified substance
*BioentityApi* | [**getSubstanceRoleAssociations**](docs/BioentityApi.md#getSubstanceRoleAssociations) | **GET** /bioentity/substance/{id}/roles | Returns associations between given drug and roles
*BioentityApi* | [**getSubstanceTreatsAssociations**](docs/BioentityApi.md#getSubstanceTreatsAssociations) | **GET** /bioentity/substance/{id}/treats | Returns substances associated with a disease
*BioentityApi* | [**getVariantCaseAssociations**](docs/BioentityApi.md#getVariantCaseAssociations) | **GET** /bioentity/variant/{id}/cases | Returns cases associated with a variant
*BioentityApi* | [**getVariantDiseaseAssociations**](docs/BioentityApi.md#getVariantDiseaseAssociations) | **GET** /bioentity/variant/{id}/diseases | Returns diseases associated with a variant
*BioentityApi* | [**getVariantGeneAssociations**](docs/BioentityApi.md#getVariantGeneAssociations) | **GET** /bioentity/variant/{id}/genes | Returns genes associated with a variant
*BioentityApi* | [**getVariantGenotypeAssociations**](docs/BioentityApi.md#getVariantGenotypeAssociations) | **GET** /bioentity/variant/{id}/genotypes | Returns genotypes associated with a variant
*BioentityApi* | [**getVariantModelAssociations**](docs/BioentityApi.md#getVariantModelAssociations) | **GET** /bioentity/variant/{id}/models | Returns models associated with a variant
*BioentityApi* | [**getVariantPhenotypeAssociations**](docs/BioentityApi.md#getVariantPhenotypeAssociations) | **GET** /bioentity/variant/{id}/phenotypes | Returns phenotypes associated with a variant
*BioentityApi* | [**getVariantPublicationAssociations**](docs/BioentityApi.md#getVariantPublicationAssociations) | **GET** /bioentity/variant/{id}/publications | Returns publications associated with a variant
*BioentitysetApi* | [**getEntitySetAssociations**](docs/BioentitysetApi.md#getEntitySetAssociations) | **GET** /bioentityset/associations | Returns compact associations for a given input set
*BioentitysetApi* | [**getEntitySetGraphResource**](docs/BioentitysetApi.md#getEntitySetGraphResource) | **GET** /bioentityset/graph | TODO Graph object spanning all entities
*BioentitysetApi* | [**getEntitySetSummary**](docs/BioentitysetApi.md#getEntitySetSummary) | **GET** /bioentityset/descriptor/counts | Summary statistics for objects associated
*BioentitysetApi* | [**getOverRepresentation**](docs/BioentitysetApi.md#getOverRepresentation) | **GET** /bioentityset/overrepresentation | Summary statistics for objects associated
*BioentitysetHomologsApi* | [**getEntitySetHomologs**](docs/BioentitysetHomologsApi.md#getEntitySetHomologs) | **GET** /bioentityset/homologs/ | Returns homology associations for a given input set of genes
*BioentitysetSlimmerApi* | [**getEntitySetAnatomySlimmer**](docs/BioentitysetSlimmerApi.md#getEntitySetAnatomySlimmer) | **GET** /bioentityset/slimmer/anatomy | For a given gene(s), summarize its annotations over a defined set of slim
*BioentitysetSlimmerApi* | [**getEntitySetFunctionSlimmer**](docs/BioentitysetSlimmerApi.md#getEntitySetFunctionSlimmer) | **GET** /bioentityset/slimmer/function | For a given gene(s), summarize its annotations over a defined set of slim
*BioentitysetSlimmerApi* | [**getEntitySetPhenotypeSlimmer**](docs/BioentitysetSlimmerApi.md#getEntitySetPhenotypeSlimmer) | **GET** /bioentityset/slimmer/phenotype | For a given gene(s), summarize its annotations over a defined set of slim
*CamApi* | [**getActivityCollection**](docs/CamApi.md#getActivityCollection) | **GET** /cam/activity | Returns list of models
*CamApi* | [**getInstanceObject**](docs/CamApi.md#getInstanceObject) | **GET** /cam/instance/{id} | Returns list of matches
*CamApi* | [**getModelCollection**](docs/CamApi.md#getModelCollection) | **GET** /cam/model | Returns list of ALL models
*CamApi* | [**getModelContributors**](docs/CamApi.md#getModelContributors) | **GET** /cam/model/contributors | Returns list of all contributors across all models
*CamApi* | [**getModelInstances**](docs/CamApi.md#getModelInstances) | **GET** /cam/instances | Returns list of all instances
*CamApi* | [**getModelObject**](docs/CamApi.md#getModelObject) | **GET** /cam/model/{id} | Returns a complete model
*CamApi* | [**getModelProperties**](docs/CamApi.md#getModelProperties) | **GET** /cam/model/properties | Returns list of all properties used across all models
*CamApi* | [**getModelPropertyValues**](docs/CamApi.md#getModelPropertyValues) | **GET** /cam/model/property_values | Returns list property-values for all models
*CamApi* | [**getModelQuery**](docs/CamApi.md#getModelQuery) | **GET** /cam/model/query | Returns list of models matching query
*CamApi* | [**getPhysicalInteraction**](docs/CamApi.md#getPhysicalInteraction) | **GET** /cam/physical_interaction | Returns list of models
*EvidenceGraphApi* | [**getEvidenceGraphObject**](docs/EvidenceGraphApi.md#getEvidenceGraphObject) | **GET** /evidence/graph/{id} | Returns evidence graph object for a given association
*EvidenceGraphApi* | [**getEvidenceGraphTable**](docs/EvidenceGraphApi.md#getEvidenceGraphTable) | **GET** /evidence/graph/{id}/table | Returns evidence as a association_results object given an association
*GenomeFeaturesApi* | [**getFeaturesWithinResource**](docs/GenomeFeaturesApi.md#getFeaturesWithinResource) | **GET** /genome/features/within/{build}/{reference}/{begin}/{end} | Returns list of matches
*GraphApi* | [**getEdgeResource**](docs/GraphApi.md#getEdgeResource) | **GET** /graph/edges/from/{id} | Returns edges emanating from a given node
*GraphApi* | [**getNodeResource**](docs/GraphApi.md#getNodeResource) | **GET** /graph/node/{id} | Returns a graph node
*IdentifierMapperApi* | [**getIdentifierMapper**](docs/IdentifierMapperApi.md#getIdentifierMapper) | **GET** /identifier/mapper/{source}/{target}/ | TODO maps a list of identifiers from a source to a target
*IdentifierPrefixesApi* | [**getPrefixCollection**](docs/IdentifierPrefixesApi.md#getPrefixCollection) | **GET** /identifier/prefixes/ | Returns list of prefixes
*IdentifierPrefixesApi* | [**getPrefixContract**](docs/IdentifierPrefixesApi.md#getPrefixContract) | **GET** /identifier/prefixes/contract/{uri} | Returns contracted URI
*IdentifierPrefixesApi* | [**getPrefixExpand**](docs/IdentifierPrefixesApi.md#getPrefixExpand) | **GET** /identifier/prefixes/expand/{id} | Returns expanded URI
*IndividualApi* | [**getIndividual**](docs/IndividualApi.md#getIndividual) | **GET** /individual/{id} | Returns list of matches
*IndividualApi* | [**getPedigree**](docs/IndividualApi.md#getPedigree) | **GET** /individual/pedigree/{id} | Returns list of matches
*MartApi* | [**getMartCaseAssociationsResource**](docs/MartApi.md#getMartCaseAssociationsResource) | **GET** /mart/case/{object_category}/{taxon} | Bulk download of case associations
*MartApi* | [**getMartDiseaseAssociationsResource**](docs/MartApi.md#getMartDiseaseAssociationsResource) | **GET** /mart/disease/{object_category}/{taxon} | Bulk download of disease associations
*MartApi* | [**getMartGeneAssociationsResource**](docs/MartApi.md#getMartGeneAssociationsResource) | **GET** /mart/gene/{object_category}/{taxon} | Bulk download of gene associations
*MartApi* | [**getMartOrthologAssociationsResource**](docs/MartApi.md#getMartOrthologAssociationsResource) | **GET** /mart/ortholog/{taxon1}/{taxon2} | Bulk download of orthologs
*MartApi* | [**getMartParalogAssociationsResource**](docs/MartApi.md#getMartParalogAssociationsResource) | **GET** /mart/paralog/{taxon1}/{taxon2} | Bulk download of paralogs
*MetadataApi* | [**getMetadataForDatasets**](docs/MetadataApi.md#getMetadataForDatasets) | **GET** /metadata/datasets | Get metadata for all datasets from SciGraph
*MmeApi* | [**postDiseaseMme**](docs/MmeApi.md#postDiseaseMme) | **POST** /mme/disease | Match a patient to diseases based on their phenotypes
*MmeApi* | [**postFlyMme**](docs/MmeApi.md#postFlyMme) | **POST** /mme/fly | Match a patient to fruit fly genes based on similar phenotypes
*MmeApi* | [**postMouseMme**](docs/MmeApi.md#postMouseMme) | **POST** /mme/mouse | Match a patient to mouse genes based on similar phenotypes
*MmeApi* | [**postNematodeMme**](docs/MmeApi.md#postNematodeMme) | **POST** /mme/nematode | Match a patient to nematode genes based on similar phenotypes
*MmeApi* | [**postZebrafishMme**](docs/MmeApi.md#postZebrafishMme) | **POST** /mme/zebrafish | Match a patient to zebrafish genes based on similar phenotypes
*NlpAnnotateApi* | [**getAnnotate**](docs/NlpAnnotateApi.md#getAnnotate) | **GET** /nlp/annotate/ | Annotate a given text using SciGraph annotator
*NlpAnnotateApi* | [**getAnnotateEntities**](docs/NlpAnnotateApi.md#getAnnotateEntities) | **GET** /nlp/annotate/entities | Annotate a given content using SciGraph annotator and get all entities from content
*NlpAnnotateApi* | [**postAnnotate**](docs/NlpAnnotateApi.md#postAnnotate) | **POST** /nlp/annotate/ | Annotate a given text using SciGraph annotator
*NlpAnnotateApi* | [**postAnnotateEntities**](docs/NlpAnnotateApi.md#postAnnotateEntities) | **POST** /nlp/annotate/entities | Annotate a given content using SciGraph annotator and get all entities from content
*OntolApi* | [**getExtractOntologySubgraphResource**](docs/OntolApi.md#getExtractOntologySubgraphResource) | **GET** /ontol/subgraph/{ontology}/{node} | Extract a subgraph from an ontology
*OntolApi* | [**getInformationContentResource**](docs/OntolApi.md#getInformationContentResource) | **GET** /ontol/information_content/{subject_category}/{object_category}/{subject_taxon} | Returns information content (IC) for a set of relevant ontology classes
*OntolApi* | [**postExtractOntologySubgraphResource**](docs/OntolApi.md#postExtractOntologySubgraphResource) | **POST** /ontol/subgraph/{ontology}/{node} | Extract a subgraph from an ontology
*OntolIdentifierApi* | [**getOntolIdentifierResource**](docs/OntolIdentifierApi.md#getOntolIdentifierResource) | **GET** /ontol/identifier/ | Fetches a map from CURIEs/IDs to labels
*OntolIdentifierApi* | [**postOntolIdentifierResource**](docs/OntolIdentifierApi.md#postOntolIdentifierResource) | **POST** /ontol/identifier/ | Fetches a map from CURIEs/IDs to labels
*OntolLabelerApi* | [**getOntolLabelerResource**](docs/OntolLabelerApi.md#getOntolLabelerResource) | **GET** /ontol/labeler/ | Fetches a map from CURIEs/IDs to labels
*OntologyApi* | [**getOntologySubset**](docs/OntologyApi.md#getOntologySubset) | **GET** /ontology/subset/{id} | Returns meta data of an ontology subset (slim)
*OntologyApi* | [**getOntologyTerm**](docs/OntologyApi.md#getOntologyTerm) | **GET** /ontology/term/{id} | Returns meta data of an ontology term
*OntologyApi* | [**getOntologyTermGraph**](docs/OntologyApi.md#getOntologyTermGraph) | **GET** /ontology/term/{id}/graph | Returns graph of an ontology term
*OntologyApi* | [**getOntologyTermSubgraph**](docs/OntologyApi.md#getOntologyTermSubgraph) | **GET** /ontology/term/{id}/subgraph | Extract a subgraph from an ontology term
*OntologyApi* | [**getOntologyTermSubsets**](docs/OntologyApi.md#getOntologyTermSubsets) | **GET** /ontology/term/{id}/subsets | Returns subsets (slims) associated to an ontology term
*OntologyApi* | [**getOntologyTermsSharedAncestor**](docs/OntologyApi.md#getOntologyTermsSharedAncestor) | **GET** /ontology/shared/{subject}/{object} | Returns the ancestor ontology terms shared by two ontology terms
*OwlOntologyApi* | [**getDlQuery**](docs/OwlOntologyApi.md#getDlQuery) | **GET** /owl/ontology/dlquery/{query} | Placeholder - use OWLery for now
*OwlOntologyApi* | [**getSparqlQuery**](docs/OwlOntologyApi.md#getSparqlQuery) | **GET** /owl/ontology/sparql/{query} | Placeholder - use direct SPARQL endpoint for now
*PairSimApi* | [**getPairSimJaccardResource**](docs/PairSimApi.md#getPairSimJaccardResource) | **GET** /pair/sim/jaccard/{id1}/{id2} | Get pairwise similarity
*RelationUsageApi* | [**getRelationUsageBetweenResource**](docs/RelationUsageApi.md#getRelationUsageBetweenResource) | **GET** /relation/usage/between/{subject_category}/{object_category} | All relations used plus count of associations
*RelationUsageApi* | [**getRelationUsagePivotLabelResource**](docs/RelationUsageApi.md#getRelationUsagePivotLabelResource) | **GET** /relation/usage/pivot/label | Relation usage count for all subj x obj category combinations, showing label
*RelationUsageApi* | [**getRelationUsagePivotResource**](docs/RelationUsageApi.md#getRelationUsagePivotResource) | **GET** /relation/usage/pivot | Relation usage count for all subj x obj category combinations
*RelationUsageApi* | [**getRelationUsageResource**](docs/RelationUsageApi.md#getRelationUsageResource) | **GET** /relation/usage/ | All relations used plus count of associations
*SearchApi* | [**getAutocomplete**](docs/SearchApi.md#getAutocomplete) | **GET** /search/entity/autocomplete/{term} | Returns list of matching concepts or entities using lexical search
*SearchApi* | [**getSearchEntities**](docs/SearchApi.md#getSearchEntities) | **GET** /search/entity/{term} | Returns list of matching concepts or entities using lexical search
*SearchApi* | [**getSearchHpoEntities**](docs/SearchApi.md#getSearchHpoEntities) | **GET** /search/entity/hpo-pl/{term} | Returns list of matching concepts or entities using lexical search
*SimApi* | [**getAnnotationScore**](docs/SimApi.md#getAnnotationScore) | **GET** /sim/score | Get annotation score
*SimApi* | [**getSimCompare**](docs/SimApi.md#getSimCompare) | **GET** /sim/compare | Compare a reference profile vs one profiles
*SimApi* | [**getSimSearch**](docs/SimApi.md#getSimSearch) | **GET** /sim/search | Search for phenotypically similar diseases or model genes
*SimApi* | [**postAnnotationScore**](docs/SimApi.md#postAnnotationScore) | **POST** /sim/score | Get annotation score
*SimApi* | [**postSimCompare**](docs/SimApi.md#postSimCompare) | **POST** /sim/compare | Compare a reference profile vs one or more profiles
*VariationSetApi* | [**deleteVariantSetItem**](docs/VariationSetApi.md#deleteVariantSetItem) | **DELETE** /variation/set/{id} | Deletes variant set
*VariationSetApi* | [**getVariantAnalyze**](docs/VariationSetApi.md#getVariantAnalyze) | **GET** /variation/set/analyze/{id} | Returns list of matches
*VariationSetApi* | [**getVariantSetItem**](docs/VariationSetApi.md#getVariantSetItem) | **GET** /variation/set/{id} | Returns a variant set
*VariationSetApi* | [**getVariantSetsArchiveCollection**](docs/VariationSetApi.md#getVariantSetsArchiveCollection) | **GET** /variation/set/archive/{year}/{month}/{day} | Returns list of variant sets from a specified time period
*VariationSetApi* | [**getVariantSetsCollection**](docs/VariationSetApi.md#getVariantSetsCollection) | **GET** /variation/set/ | Returns list of variant sets
*VariationSetApi* | [**postVariantSetsCollection**](docs/VariationSetApi.md#postVariantSetsCollection) | **POST** /variation/set/ | Creates a new variant set
*VariationSetApi* | [**putVariantSetItem**](docs/VariationSetApi.md#putVariantSetItem) | **PUT** /variation/set/{id} | Updates a variant set


## Documentation for Models

 - [APageOfResults](docs/APageOfResults.md)
 - [AbstractPropertyValue](docs/AbstractPropertyValue.md)
 - [AnnotationExtension](docs/AnnotationExtension.md)
 - [Association](docs/Association.md)
 - [AssociationResults](docs/AssociationResults.md)
 - [AutocompleteResult](docs/AutocompleteResult.md)
 - [AutocompleteResults](docs/AutocompleteResults.md)
 - [BioObject](docs/BioObject.md)
 - [BioObjectCore](docs/BioObjectCore.md)
 - [CompactAssociationSet](docs/CompactAssociationSet.md)
 - [CompareInput](docs/CompareInput.md)
 - [D2PAssociation](docs/D2PAssociation.md)
 - [D2PAssociationResults](docs/D2PAssociationResults.md)
 - [Edge](docs/Edge.md)
 - [EntityAnnotationResult](docs/EntityAnnotationResult.md)
 - [EntityReference](docs/EntityReference.md)
 - [Feature](docs/Feature.md)
 - [Graph](docs/Graph.md)
 - [IcNode](docs/IcNode.md)
 - [LayAutocomplete](docs/LayAutocomplete.md)
 - [LayResults](docs/LayResults.md)
 - [NamedObject](docs/NamedObject.md)
 - [NamedObjectCore](docs/NamedObjectCore.md)
 - [Node](docs/Node.md)
 - [PageOfVariantSets](docs/PageOfVariantSets.md)
 - [PairwiseMatch](docs/PairwiseMatch.md)
 - [Relation](docs/Relation.md)
 - [RelationRef](docs/RelationRef.md)
 - [SearchResult](docs/SearchResult.md)
 - [Seq](docs/Seq.md)
 - [SequenceFeature](docs/SequenceFeature.md)
 - [SequenceLocation](docs/SequenceLocation.md)
 - [SequencePosition](docs/SequencePosition.md)
 - [SimMatch](docs/SimMatch.md)
 - [SimMatchAllOfPairwiseMatch](docs/SimMatchAllOfPairwiseMatch.md)
 - [SimMetadata](docs/SimMetadata.md)
 - [SimQuery](docs/SimQuery.md)
 - [SimResult](docs/SimResult.md)
 - [Span](docs/Span.md)
 - [SufficiencyOutput](docs/SufficiencyOutput.md)
 - [SufficiencyPostInput](docs/SufficiencyPostInput.md)
 - [SynonymPropertyValue](docs/SynonymPropertyValue.md)
 - [Taxon](docs/Taxon.md)
 - [Token](docs/Token.md)
 - [TypedNode](docs/TypedNode.md)
 - [VariantSet](docs/VariantSet.md)


<a id="documentation-for-authorization"></a>
## Documentation for Authorization

Endpoints do not require authorization.


## Recommendation

It's recommended to create an instance of `ApiClient` per thread in a multithreaded environment to avoid any potential issues.

## Author



